erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -503,11 +503,14 @@ def validate_auto_set_posting_time(self):
 
 	def so_dn_required(self):
 		"""check in manage account if sales order / delivery note required or not."""
+		if self.is_return:
+			return
 		dic = {'Sales Order':['so_required', 'is_pos'],'Delivery Note':['dn_required', 'update_stock']}
 		for i in dic:
 			if frappe.db.get_single_value('Selling Settings', dic[i][0]) == 'Yes':
 				for d in self.get('items'):
-					if (d.item_code and frappe.get_cached_value('Item', d.item_code, 'is_stock_item') == 1
+					cached_item_code = frappe.get_cached_value('Item', d.item_code, 'is_stock_item')
```

Comment:
Variable naming is too bad. it should define the purpose like `is_stock_item`

Reply:
done